axiom/nr_app_harvest.c

```diff
@@ -12,6 +12,21 @@
 #include "nr_app_harvest_private.h"
 #include "util_logging.h"
 
+/*
+ * The build pipeline for PHP 8.2 uses a Debian image that is built on a
+ * version of glibc that contains an newer glibc version symbol for `pow()`
+ * which is incompatible with older distributions, causing RPM installs to fail.
+ *
+ * To pin the version of `pow` to a version compatible with all NR PHP Agent
+ * supported OSes, we utilize `asm()` and `.symver` to instruct the linker to
+ * select an older version- in this case, 2.17.
+ */
+#if defined(__GLIBC__)
+#if ((__GLIBC__) >= 2) && ((__GLIBC_MINOR__) > 17)
+__asm__(".symver pow,pow@GLIBC_2.17");
```

Comment:
libm does not provide the same versions of a symbol across different architectures.
x86_64 has 2.2.5 and 2.29:
```
root@fa2762fd1382:/# objdump -T /lib/x86_64-linux-gnu/libm-2.31.so | grep -w pow
0000000000040040 g    DF .text	00000000000000b6  GLIBC_2.29  pow
0000000000010020 g    DF .text	000000000000014a (GLIBC_2.2.5) pow
root@fa2762fd1382:/# uname -m
x86_64
root@fa2762fd1382:/# grep PRETTY /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
```
aarch64 has 2.17 and 2.29:
```
root@6d553b8b57f8:/# objdump -T /lib/aarch64-linux-gnu/libm-2.31.so | grep -w pow
0000000000035ff0 g    DF .text	000000000000044c  GLIBC_2.29  pow
000000000000ebb0 g    DF .text	000000000000013c (GLIBC_2.17) pow
root@6d553b8b57f8:/# uname -m
aarch64
root@6d553b8b57f8:/# grep PRETTY /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
```

```suggestion
#if defined (__x86_64__)
  __asm__(".symver pow,pow@GLIBC_2.2.5");
#elif defined (__aarch64__)
  __asm__(".symver pow,pow@GLIBC_2.17");
#endif
```